format.sh

```diff
@@ -129,6 +129,9 @@ if [[ "$1" == '--files' ]]; then
     pylint "${PYLINT_FLAGS[@]}" "${@:2}"
 elif [[ "$1" == '--all' ]]; then
     # Pylint entire sky directory.
+    pylint "${PYLINT_FLAGS[@]}" $(git ls-files 'sky/*.py' 'sky/*.pyi')
+elif [[ "$1" == '--legacy' ]]; then
```

Comment:
Could you make sure that this flag is handled in the yapf logic as well?

Reply:
Oh, I forgot to handle the `--legacy` flag for `yapf`. If `yapf` can ensure full coverage, then the modification should look like this?

```sh
## This flag formats individual files. --files *must* be the first command line
## arg to use this option.
if [[ "$1" == '--files' ]]; then
   format "${@:2}"
   # If `--all` is passed, then any further arguments are ignored and the
   # entire python directory is formatted.
elif [[ "$1" == '--all' ]]; then
   format_all
elif [[ "$1" == '--legacy' ]]; then
   format_all
else
   # Format only the files that changed in last commit.
   format_changed
fi
echo 'SkyPilot yapf: Done'
```

The reason for this modification is that `pylint` previously used `--all` but couldn’t check all files due to its own limitations. If `yapf` can check all files when `--all` is specified originally, then in the new version, the behavior for `--legacy` and `--all` should be identical. Do you think that would work?